# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.h
# Commit: f38b9eb31da7
# Full Hash: f38b9eb31da7616b63552abb364e9bd374ee22ce
# Author: Cameron McCormack <cam@mcc.id.au>
# Date: 2019-06-07 09:47:19
# Regressor Bug: 1490792
# File Overlap Count: 1
# Description:
#   Bug 1490792 - Part 3: Split some work out of LoadPlatformFont. r=jfkthame
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D33896
# ==============================================================================

diff -r ebecfbeb55c6 -r f38b9eb31da7 gfx/thebes/gfxUserFontSet.h
--- a/gfx/thebes/gfxUserFontSet.h	Thu Jun 06 18:23:29 2019 +0000
+++ b/gfx/thebes/gfxUserFontSet.h	Thu Jun 06 18:23:31 2019 +0000
@@ -661,7 +661,13 @@
   // returns true if platform font creation successful
   // Ownership of aFontData is passed in here; the font must
   // ensure that it is eventually deleted with free().
-  bool LoadPlatformFont(const uint8_t* aFontData, uint32_t aLength);
+  bool LoadPlatformFontSync(const uint8_t* aFontData, uint32_t aLength);
+
+  // helper method for LoadPlatformFontSync
+  bool LoadPlatformFont(const uint8_t* aOriginalFontData,
+                        uint32_t aOriginalLength, gfxUserFontType aFontType,
+                        const uint8_t* aSanitizedFontData,
+                        uint32_t aSanitizedLength);
 
   // store metadata and src details for current src into aFontEntry
   void StoreUserFontData(gfxFontEntry* aFontEntry, bool aPrivate,
